src/Raven.Server/Rachis/Candidate.cs

```diff
@@ -150,12 +151,19 @@ private void Run()
                             StateChange();
 
                             var connections = new Dictionary<string, RemoteConnection>();
+                            var versions = new List<int>();
                             foreach (var candidateAmbassador in _voters)
                             {
                                 connections[candidateAmbassador.Tag] = candidateAmbassador.Connection;
+                                if (candidateAmbassador.ClusterCommandsVersion > 0)
```

Comment:
If this is 0, that is also information, it means the client is too old

Reply:
You need to distinguish between "don't know yet (other node is down)" to "The other node is too old to tell me"

Reply:
OK, in any case I think we need to use a median here instead of taking the minimum of the voters.

Reply:
Actually, the distinguish between old version and "don't know" happened when the candidate ambassador reads the 'RequestVoteResponse'. 
https://github.com/karmeli87/ravendb/blob/7eb891a9b7e09cda1b301160fe7cd3317a821fe7/src/Raven.Server/Rachis/RequestVote.cs#L29

Reply:
And it is at that you can take this into account.
But we need the minimum number among the majority that we know of.